syntax = "proto3";
package ezio;

message Torrent {
	// hex sha1 hash lowercase
	string hash = 1;
	// torrent name
	string name = 2;
	// float point number bewteen [0, 1]
	double progress = 3;
	// download rate byte per second
	int32 download = 4;
	// upload rate byte per second
	int32 upload = 5;
	// active time
	int32 active_time = 6;
}

message UpdateRequest {
	// update status only in hashs, if empty update all
	// lowercase hex sha1 hash
	repeated string hashes = 1;
}

message UpdateStatus {
	// lowercase hex sha1 hash
	repeated string hashes = 1;
	map<string, Torrent> torrents = 2;
}

service EZIO {
	rpc GetTorrentStatus(UpdateRequest) returns (UpdateStatus) {}
}
